src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1484,7 +1484,7 @@ private static bool WrongArity(Symbol symbol, int arity, bool diagnose, LookupOp
                         NamedTypeSymbol namedType = (NamedTypeSymbol)symbol;
                         // non-declared types only appear as using aliases (aliases are arity 0)
                         Debug.Assert(object.ReferenceEquals(namedType.ConstructedFrom, namedType));
-                        if (namedType.Arity != arity || options.IsAttributeTypeLookup() && arity != 0)
+                        if (namedType.Arity != arity || (options.IsAttributeTypeLookup() && !compilation.LanguageVersion.AllowGenericAttributes()) && arity != 0)
```

Comment:
I think we have a similar problem here. The LangVer should not affect the viability (`WrongArity` is used in `CheckViability`). It should only affect diagnostics.

Reply:
fixed